src/main/java/com/api/entity/SxCompositeFairing.java

```diff
@@ -2,17 +2,34 @@
 
 import com.api.dto.DimensionRocket;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 
 @AllArgsConstructor
 @NoArgsConstructor
 @Getter
+@Entity
+@Table(name="sx_composite_fairing")
 public class SxCompositeFairing {
+    @Id
```

Comment:
never use same objects for fetching and storing, every operation should have thei objects, please create a separate set of JPA object for the db, SxCompositeFairingJpa for example